lapack/gonum/dgtsv.go

```diff
@@ -0,0 +1,100 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gonum
+
+import "math"
+
+// Dgtsv solves the equation
+//  A * X = B
+// where A is an n×n tridiagonal matrix. It uses Gaussian elimination with
+// partial pivoting. The equation Aᵀ * X = B may be solved by swapping the
+// arguments for du and dl.
+//
+// On entry, dl, d and du contain the sub-diagonal, the diagonal and the
+// super-diagonal, respectively, of A. On return, the first n-2 elements of dl
+// will be overwritten by the n-2 elements of the second super diagonal of the
+// upper triangular matrix U from the LU factorization of A. d will be
+// overwritten by the diagonal of U and du by the first super-diagonal of U.
+//
+// On entry, b contains the n×nrhs right-hand side matrix B. On return, if ok is
+// true, it will be overwritten by the solution matrix X.
+//
+// Dgtsv returns whether the solution X has been successfuly computed.
+func (impl Implementation) Dgtsv(n, nrhs int, dl, d, du []float64, b []float64, ldb int) (ok bool) {
+	switch {
+	case n < 0:
+		panic(nLT0)
+	case nrhs < 0:
+		panic(nrhsLT0)
+	case ldb < max(1, nrhs):
+		panic(badLdB)
+	}
+
+	if n == 0 || nrhs == 0 {
+		return true
```

Comment:
Add the `nhrs == 0` case of the back solve?

Reply:
I thought about that but `b` is already accessed during the factorization. And we check the length of b too, so the order would be: access b during the factorization (even though it's not actually accessed when `nrhs == 0`, quick return if `nrhs == 0`, and only then check length of b which I think is a weird order. And all this just so that the caller has U available (in a documented way).